### ISSUE TITLE
swc: Type-only import left in JS emit

### ISSUE THREAD

**Issue by nayeemrmn** [denoland/deno#10462]
`temp.ts`:
```ts
import { foo } from "./temp2.ts"; // This import statement is wrongly left in the emit.

const _: foo = null;
console.log({ foo: 1 }); // This usage of the name `foo` wrongly counts as a value-usage of the imported `foo`.
```
`temp2.ts`:
```ts
export type foo = unknown;
```

cc @kdy1

**Comment by kdy1** [comment]
Fixed on the upstream.

**Comment by nayeemrmn** [comment]
Fixed by #10559.
